lib/encrypt.c

```diff
@@ -65,7 +65,8 @@
 		(void) fprintf (shadow_logfd,
 		                _("crypt method not supported by libcrypt? (%s)\n"),
 		                method);
-		exit (EXIT_FAILURE);
+		errno = EINVAL;
+		return NULL;
```

Comment:
Did you check that all callers will handle this change correctly?

Reply:
Yes. Especially since they already have to if crypt returns NULL.